engine/src/main/java/com/sandboni/core/engine/Processor.java

```diff
@@ -115,10 +114,14 @@ private boolean isIntegrationStage() {
     }
 
     public ChangeScope<Change> getScope() {
+        log.info("[{}] Getting change scope", Thread.currentThread().getName());
+        long start = System.nanoTime();
         try {
             return changeDetector.getChanges(arguments.getFromChangeId(), arguments.getToChangeId());
         } catch (SourceControlException e) {
             throw new ParseRuntimeException(e);
+        } finally {
+            log.info("[{}] Change scope retrieved in {} milliseconds", Thread.currentThread().getName(), elapsedTime(start));
```

Comment:
Maybe we should change the message a bit, because with current implementation we might print "Change scope retrieved.." even though there was an exception thrown..

Reply:
Done.